File: Leetcode/128/Solution.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s1 = set(nums)
        longest = 0

        for i in nums:
            if i - 1 not in s1:
                length = 0
                while i + length in s1:
                    length += 1
                longest = max(length, longest)
        
        return longest

File: Leetcode/128/Solution.java
import java.util.HashSet;

class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        for (int num : nums) {
            set.add(num);
        }

        int longest = 0;

        for (int num : set) {
            // Only start counting if num is the start of a sequence
            if (!set.contains(num - 1)) {
                int currentNum = num;
                int count = 1;

                while (set.contains(currentNum + 1)) {
                    currentNum++;
                    count++;
                }

                longest = Math.max(longest, count);
            }
        }

        return longest;
    }

    public static void main(String[] args) {
        Solution s1 = new Solution();
        System.out.println(s1.longestConsecutive(new int[] { 100, 4, 200, 1, 3, 2 }));
    }
}
